.github/workflows/rust.yml

```
@@ -0,0 +1,43 @@
+name: Rust
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  linux:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+      - run: cargo test --workspace --all-features
+        env:
+          RUST_BACKTRACE: 1
+
+  windows:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+      - run: cargo test --workspace --all-features
+        env:
+          RUST_BACKTRACE: 1
+
+  macos:
+    runs-on:  macos-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+      - run: cargo test --workspace --all-features
+        env:
+          RUST_BACKTRACE: 1
```

FYI, this could be reduced to one matrix-based job (following DRY principal):

```suggestion
  test:
    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest, macos-latest]
    runs-on: ${{ matrix.os }}
    steps:
      - uses: actions/checkout@v4
      - uses: dtolnay/rust-toolchain@stable
      - run: cargo test --workspace --all-features
        env:
          RUST_BACKTRACE: 1
```
More detail in [github docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow).

---
PS - For security and maintenance reasons, I prefer to not rely on third-party actions (those that are not [maintained by github](https://github.com/actions/)). All [GitHub runner images](https://github.com/actions/runner-images?tab=readme-ov-file#available-images) ship with rustup installed. So, you could also
```diff
-      - uses: dtolnay/rust-toolchain@stable
+      - run: rustup update --no-self-update
```